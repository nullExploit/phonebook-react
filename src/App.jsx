import "./App.css";
import { createBrowserRouter, RouterProvider } from "react-router-dom";
import PhoneBookBox from "./components/PhoneBookBox";
import ErrorPage from "./components/ErrorPage";
import PhoneBookAdd from "./components/PhoneBookAdd";
import Providers from "./lib/redux/Provider";

const router = createBrowserRouter([
  {
    path: "/",
    element: <PhoneBookBox />,
    errorElement: <ErrorPage />,
  },
  {
    path: "/add",
    element: <PhoneBookAdd />,
  },
]);

function App() {
  return (
    <Providers>
      <RouterProvider router={router} />
    </Providers>
  );
}

export default App;
