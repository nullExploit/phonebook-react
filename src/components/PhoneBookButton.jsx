import {
  faCheck,
  faFloppyDisk,
  faPenToSquare,
  faTrashCan,
  faXmark,
} from "@fortawesome/free-solid-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { useDispatch } from "react-redux";
import { useState } from "react";
import {
  deletePhoneBook,
  editPhoneBook,
} from "../lib/redux/phonebooks/PhoneBookSlice";

export default function PhoneBookButton({ name, phone, editStatus, edit, id }) {
  const dispatch = useDispatch();

  const [valid, setValid] = useState(false);

  if (valid) {
    return (
      <div className="button">
        <FontAwesomeIcon
          className="buttons"
          onClick={() => {
            dispatch(deletePhoneBook(id));
            setValid((prev) => !prev);
          }}
          icon={faCheck}
        />
        <FontAwesomeIcon
          className="buttons"
          onClick={() => setValid((prev) => !prev)}
          icon={faXmark}
        />
      </div>
    );
  }

  if (editStatus) {
    return (
      <div className="button">
        <FontAwesomeIcon
          className="buttons"
          onClick={() => {
            dispatch(editPhoneBook(id, name, phone));
            edit();
          }}
          icon={faFloppyDisk}
        />
      </div>
    );
  }

  return (
    <div className="button">
      <FontAwesomeIcon
      aria-label="editbutton"
        className="buttons"
        onClick={() => edit()}
        icon={faPenToSquare}
      />
      <FontAwesomeIcon
        className="buttons"
        onClick={() => setValid((prev) => !prev)}
        icon={faTrashCan}
      />
    </div>
  );
}
