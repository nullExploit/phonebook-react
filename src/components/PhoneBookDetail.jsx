import { useState } from "react";
import PhoneBookButton from "./PhoneBookButton";

export default function PhoneBookDetail({ id, name, phone }) {
  const [edit, setEdit] = useState(false);
  const [tempName, setTempName] = useState(name);
  const [tempPhone, setTempPhone] = useState(phone);

  const onEdit = () => {
    setEdit((prev) => !prev);
  };

  if (edit) {
    return (
      <div className="details">
        <input
          type="text"
          className="edits"
          value={tempName}
          onChange={(evt) => setTempName(evt.target.value)}
        />
        <input
          type="text"
          className="edits"
          value={tempPhone}
          onChange={(evt) => setTempPhone(evt.target.value)}
        />
        <br />
        <PhoneBookButton
          edit={onEdit}
          name={tempName}
          phone={tempPhone}
          editStatus={edit}
          id={id}
        />
      </div>
    );
  }

  return (
    <div className="details">
      <p className="name">{name}</p>
      <p>{phone}</p>
      <br />
      <PhoneBookButton
        edit={onEdit}
        name={tempName}
        phone={tempPhone}
        editStatus={edit}
        id={id}
      />
    </div>
  );
}
