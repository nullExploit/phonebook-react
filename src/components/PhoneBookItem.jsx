import { useState } from "react";
import PhoneBookDetail from "./PhoneBookDetail";
import { updateAvatarPhoneBookAsync } from "../lib/redux/phonebooks/PhoneBookSlice";
import { useDispatch } from "react-redux";

export default function PhoneBookItem({ id, name, phone, avatar }) {
  const [fileName, setFileName] = useState();
  const [file, setFile] = useState();
  const dispatch = useDispatch();

  const handleSubmit = async (event, id, file) => {
    event.preventDefault();
    dispatch(updateAvatarPhoneBookAsync({ id, data: file }));
  };

  const handleChange = (event, setFileName, setFile, id) => {
    if (event.target.files[0]) {
      setFileName(URL.createObjectURL(event.target.files[0]));
      setFile(event.target.files[0]);
      setTimeout(() => {
        document.getElementById(`submit${id}`).click();
      });
    }
  };

  return (
    <div className="phonebook-item">
      <img
        className="img"
        src={fileName ? fileName : avatar}
        alt={file}
        onClick={() => document.getElementById(`fileUpload${id}`).click()}
      />
      <form
        onSubmit={(evt) => handleSubmit(evt, id, file)}
        id={`formUpload${id}`}
        style={{ display: "none" }}
        encType="multipart/form-data"
      >
        <input
          type="file"
          id={`fileUpload${id}`}
          accept="image/*"
          onChange={(evt) => handleChange(evt, setFileName, setFile, id)}
        />
        <button type="submit" id={`submit${id}`} />
      </form>
      <PhoneBookDetail name={name} phone={phone} id={id} />
    </div>
  );
}
