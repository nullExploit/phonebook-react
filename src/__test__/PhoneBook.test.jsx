import { fireEvent, render, screen } from "@testing-library/react";
import Providers from "../lib/redux/Provider";
import { RouterProvider } from "react-router-dom";
import { router } from "../App";
import PhoneBookButton from "../components/PhoneBookButton";
import PhoneBookItem from "../components/PhoneBookItem";
import PhoneBookDetail from "../components/PhoneBookDetail";

describe("Should render all phonebook correctly", () => {
  it("Should render phonebook button correctly", () => {
    const setOnEdit = jest.fn();

    const el = (stat) => (
      <Providers>
        <PhoneBookButton edit={setOnEdit} editStatus={stat} />
      </Providers>
    );

    const { baseElement, rerender } = render(el(false));

    expect(baseElement).toMatchSnapshot();
    fireEvent.click(screen.getByLabelText("editbutton"));
    rerender(el(true));
    expect(screen.getByLabelText("editbutton").getAttribute("data-icon")).toBe(
      "floppy-disk"
    );
    expect(setOnEdit).toHaveReturnedTimes(1);
    expect(screen.getByLabelText("editbutton").parentElement.className).toBe(
      "button"
    );
    rerender(el(false));
    fireEvent.click(document.querySelectorAll(".buttons")[1]);
    expect(
      document.querySelectorAll(".buttons")[0].getAttribute("data-icon")
    ).toBe("check");
    expect(
      document.querySelectorAll(".buttons")[1].getAttribute("data-icon")
    ).toBe("xmark");
    fireEvent.click(document.querySelectorAll(".buttons")[1]);
  });

  it("Should render phonebook detail correctly", () => {
    const { baseElement } = render(
      <Providers>
        <PhoneBookDetail id={1} name="TESTING" phone="0911111111" />
      </Providers>
    );
    expect(baseElement).toMatchSnapshot();
    expect(screen.getByText("TESTING").tagName).toBe("P");
    expect(screen.getByText("0911111111").tagName).toBe("P");
    fireEvent.click(screen.getByLabelText("editbutton"));
    fireEvent.change(document.querySelectorAll("input")[0], {
      target: { value: "TESTING" },
    });
    fireEvent.change(document.querySelectorAll("input")[1], {
      target: { value: "08123123" },
    });
    expect(document.querySelectorAll("input")[0].value).toBe("TESTING");
    expect(document.querySelectorAll("input")[1].value).toBe("08123123");
    fireEvent.click(screen.getByLabelText("editbutton"));
    fireEvent.click(document.querySelectorAll(".buttons")[1]);
    expect(
      document.querySelectorAll(".buttons")[0].getAttribute("data-icon")
    ).toBe("check");
    expect(
      document.querySelectorAll(".buttons")[1].getAttribute("data-icon")
    ).toBe("xmark");
    fireEvent.click(document.querySelectorAll(".buttons")[1]);
  });

  it("Should render phonebook item correctly", () => {
    const { baseElement } = render(
      <Providers>
        <PhoneBookItem id={1} name="TESTING" phone="0911111111" avatar={null} />
      </Providers>
    );
    expect(baseElement).toMatchSnapshot();
    expect(screen.getByText("TESTING").tagName).toBe("P");
    expect(screen.getByText("0911111111").tagName).toBe("P");
    expect(
      screen
        .getByText("TESTING")
        .parentElement.parentElement.querySelector(".img").src
    ).toBe("");
    expect(
      screen
        .getByText("TESTING")
        .parentElement.parentElement.querySelector("form").enctype
    ).toBe("multipart/form-data");
    expect(
      screen
        .getByText("TESTING")
        .parentElement.parentElement.querySelector("form").id
    ).toBe("formUpload1");
    expect(
      screen
        .getByText("TESTING")
        .parentElement.parentElement.querySelector("input").type
    ).toBe("file");
    expect(
      screen
        .getByText("TESTING")
        .parentElement.parentElement.querySelector("input").id
    ).toBe("fileUpload1");
    expect(
      screen
        .getByText("TESTING")
        .parentElement.parentElement.querySelector("input").accept
    ).toBe("image/*");
    expect(
      screen
        .getByText("TESTING")
        .parentElement.parentElement.querySelector("button").type
    ).toBe("submit");
    expect(
      screen
        .getByText("TESTING")
        .parentElement.parentElement.querySelector("button").id
    ).toBe("submit1");
    fireEvent.click(screen.getByLabelText("editbutton"));
    fireEvent.change(document.querySelectorAll("input[type='text']")[0], {
      target: { value: "TESTING" },
    });
    fireEvent.change(document.querySelectorAll("input[type='text']")[1], {
      target: { value: "08123123" },
    });
    expect(document.querySelectorAll("input[type='text']")[0].value).toBe(
      "TESTING"
    );
    expect(document.querySelectorAll("input[type='text']")[1].value).toBe(
      "08123123"
    );
    fireEvent.click(screen.getByLabelText("editbutton"));
    fireEvent.click(document.querySelectorAll(".buttons")[1]);
    expect(
      document.querySelectorAll(".buttons")[0].getAttribute("data-icon")
    ).toBe("check");
    expect(
      document.querySelectorAll(".buttons")[1].getAttribute("data-icon")
    ).toBe("xmark");
    fireEvent.click(document.querySelectorAll(".buttons")[1]);
  });

  it("Should render view correctly", () => {
    const { baseElement } = render(
      <Providers>
        <RouterProvider router={router} />
      </Providers>
    );
    expect(baseElement).toMatchSnapshot();
    fireEvent.change(document.querySelector(".search-bar"), {
      target: { value: "TEST" },
    });
    expect(screen.getByDisplayValue("TEST").value).toBe("TEST");
    fireEvent.change(document.querySelector(".search-bar"), {
      target: { value: "TES" },
    });
    expect(screen.getByDisplayValue("TES").value).toBe("TES");
  });
});
