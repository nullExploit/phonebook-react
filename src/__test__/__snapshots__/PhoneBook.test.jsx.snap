// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render all phonebook correctly Should render phonebook button correctly 1`] = `
<body>
  <div>
    <div
      class="button"
    >
      <svg
        aria-hidden="true"
        aria-label="editbutton"
        class="svg-inline--fa fa-pen-to-square buttons"
        data-icon="pen-to-square"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M471.6 21.7c-21.9-21.9-57.3-21.9-79.2 0L362.3 51.7l97.9 97.9 30.1-30.1c21.9-21.9 21.9-57.3 0-79.2L471.6 21.7zm-299.2 220c-6.1 6.1-10.8 13.6-13.5 21.9l-29.6 88.8c-2.9 8.6-.6 18.1 5.8 24.6s15.9 8.7 24.6 5.8l88.8-29.6c8.2-2.7 15.7-7.4 21.9-13.5L437.7 172.3 339.7 74.3 172.4 241.7zM96 64C43 64 0 107 0 160L0 416c0 53 43 96 96 96l256 0c53 0 96-43 96-96l0-96c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 96c0 17.7-14.3 32-32 32L96 448c-17.7 0-32-14.3-32-32l0-256c0-17.7 14.3-32 32-32l96 0c17.7 0 32-14.3 32-32s-14.3-32-32-32L96 64z"
          fill="currentColor"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-trash-can buttons"
        data-icon="trash-can"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M135.2 17.7C140.6 6.8 151.7 0 163.8 0L284.2 0c12.1 0 23.2 6.8 28.6 17.7L320 32l96 0c17.7 0 32 14.3 32 32s-14.3 32-32 32L32 96C14.3 96 0 81.7 0 64S14.3 32 32 32l96 0 7.2-14.3zM32 128l384 0 0 320c0 35.3-28.7 64-64 64L96 512c-35.3 0-64-28.7-64-64l0-320zm96 64c-8.8 0-16 7.2-16 16l0 224c0 8.8 7.2 16 16 16s16-7.2 16-16l0-224c0-8.8-7.2-16-16-16zm96 0c-8.8 0-16 7.2-16 16l0 224c0 8.8 7.2 16 16 16s16-7.2 16-16l0-224c0-8.8-7.2-16-16-16zm96 0c-8.8 0-16 7.2-16 16l0 224c0 8.8 7.2 16 16 16s16-7.2 16-16l0-224c0-8.8-7.2-16-16-16z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</body>
`;

exports[`Should render all phonebook correctly Should render phonebook detail correctly 1`] = `
<body>
  <div>
    <div
      class="details"
    >
      <p
        class="name"
      >
        TESTING
      </p>
      <p>
        0911111111
      </p>
      <br />
      <div
        class="button"
      >
        <svg
          aria-hidden="true"
          aria-label="editbutton"
          class="svg-inline--fa fa-pen-to-square buttons"
          data-icon="pen-to-square"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M471.6 21.7c-21.9-21.9-57.3-21.9-79.2 0L362.3 51.7l97.9 97.9 30.1-30.1c21.9-21.9 21.9-57.3 0-79.2L471.6 21.7zm-299.2 220c-6.1 6.1-10.8 13.6-13.5 21.9l-29.6 88.8c-2.9 8.6-.6 18.1 5.8 24.6s15.9 8.7 24.6 5.8l88.8-29.6c8.2-2.7 15.7-7.4 21.9-13.5L437.7 172.3 339.7 74.3 172.4 241.7zM96 64C43 64 0 107 0 160L0 416c0 53 43 96 96 96l256 0c53 0 96-43 96-96l0-96c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 96c0 17.7-14.3 32-32 32L96 448c-17.7 0-32-14.3-32-32l0-256c0-17.7 14.3-32 32-32l96 0c17.7 0 32-14.3 32-32s-14.3-32-32-32L96 64z"
            fill="currentColor"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-trash-can buttons"
          data-icon="trash-can"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M135.2 17.7C140.6 6.8 151.7 0 163.8 0L284.2 0c12.1 0 23.2 6.8 28.6 17.7L320 32l96 0c17.7 0 32 14.3 32 32s-14.3 32-32 32L32 96C14.3 96 0 81.7 0 64S14.3 32 32 32l96 0 7.2-14.3zM32 128l384 0 0 320c0 35.3-28.7 64-64 64L96 512c-35.3 0-64-28.7-64-64l0-320zm96 64c-8.8 0-16 7.2-16 16l0 224c0 8.8 7.2 16 16 16s16-7.2 16-16l0-224c0-8.8-7.2-16-16-16zm96 0c-8.8 0-16 7.2-16 16l0 224c0 8.8 7.2 16 16 16s16-7.2 16-16l0-224c0-8.8-7.2-16-16-16zm96 0c-8.8 0-16 7.2-16 16l0 224c0 8.8 7.2 16 16 16s16-7.2 16-16l0-224c0-8.8-7.2-16-16-16z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
  </div>
</body>
`;

exports[`Should render all phonebook correctly Should render phonebook item correctly 1`] = `
<body>
  <div>
    <div
      class="phonebook-item"
    >
      <img
        class="img"
      />
      <form
        enctype="multipart/form-data"
        id="formUpload1"
        style="display: none;"
      >
        <input
          accept="image/*"
          id="fileUpload1"
          type="file"
        />
        <button
          id="submit1"
          type="submit"
        />
      </form>
      <div
        class="details"
      >
        <p
          class="name"
        >
          TESTING
        </p>
        <p>
          0911111111
        </p>
        <br />
        <div
          class="button"
        >
          <svg
            aria-hidden="true"
            aria-label="editbutton"
            class="svg-inline--fa fa-pen-to-square buttons"
            data-icon="pen-to-square"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M471.6 21.7c-21.9-21.9-57.3-21.9-79.2 0L362.3 51.7l97.9 97.9 30.1-30.1c21.9-21.9 21.9-57.3 0-79.2L471.6 21.7zm-299.2 220c-6.1 6.1-10.8 13.6-13.5 21.9l-29.6 88.8c-2.9 8.6-.6 18.1 5.8 24.6s15.9 8.7 24.6 5.8l88.8-29.6c8.2-2.7 15.7-7.4 21.9-13.5L437.7 172.3 339.7 74.3 172.4 241.7zM96 64C43 64 0 107 0 160L0 416c0 53 43 96 96 96l256 0c53 0 96-43 96-96l0-96c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 96c0 17.7-14.3 32-32 32L96 448c-17.7 0-32-14.3-32-32l0-256c0-17.7 14.3-32 32-32l96 0c17.7 0 32-14.3 32-32s-14.3-32-32-32L96 64z"
              fill="currentColor"
            />
          </svg>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-trash-can buttons"
            data-icon="trash-can"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M135.2 17.7C140.6 6.8 151.7 0 163.8 0L284.2 0c12.1 0 23.2 6.8 28.6 17.7L320 32l96 0c17.7 0 32 14.3 32 32s-14.3 32-32 32L32 96C14.3 96 0 81.7 0 64S14.3 32 32 32l96 0 7.2-14.3zM32 128l384 0 0 320c0 35.3-28.7 64-64 64L96 512c-35.3 0-64-28.7-64-64l0-320zm96 64c-8.8 0-16 7.2-16 16l0 224c0 8.8 7.2 16 16 16s16-7.2 16-16l0-224c0-8.8-7.2-16-16-16zm96 0c-8.8 0-16 7.2-16 16l0 224c0 8.8 7.2 16 16 16s16-7.2 16-16l0-224c0-8.8-7.2-16-16-16zm96 0c-8.8 0-16 7.2-16 16l0 224c0 8.8 7.2 16 16 16s16-7.2 16-16l0-224c0-8.8-7.2-16-16-16z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Should render all phonebook correctly Should render view correctly 1`] = `
<body>
  <div>
    <section>
      <div
        class="search-container"
      >
        <button
          class="sorting-button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-arrow-up-z-a "
            data-icon="arrow-up-z-a"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M183.6 42.4C177.5 35.8 169 32 160 32s-17.5 3.8-23.6 10.4l-88 96c-11.9 13-11.1 33.3 2 45.2s33.3 11.1 45.2-2L128 146.3 128 448c0 17.7 14.3 32 32 32s32-14.3 32-32l0-301.7 32.4 35.4c11.9 13 32.2 13.9 45.2 2s13.9-32.2 2-45.2l-88-96zM320 64c0 17.7 14.3 32 32 32l50.7 0-73.4 73.4c-9.2 9.2-11.9 22.9-6.9 34.9s16.6 19.8 29.6 19.8l128 0c17.7 0 32-14.3 32-32s-14.3-32-32-32l-50.7 0 73.4-73.4c9.2-9.2 11.9-22.9 6.9-34.9s-16.6-19.8-29.6-19.8L352 32c-17.7 0-32 14.3-32 32zm96 192c-12.1 0-23.2 6.8-28.6 17.7l-64 128-16 32c-7.9 15.8-1.5 35 14.3 42.9s35 1.5 42.9-14.3l7.2-14.3 88.4 0 7.2 14.3c7.9 15.8 27.1 22.2 42.9 14.3s22.2-27.1 14.3-42.9l-16-32-64-128C439.2 262.8 428.1 256 416 256zM395.8 400L416 359.6 436.2 400l-40.4 0z"
              fill="currentColor"
            />
          </svg>
        </button>
        <input
          class="search-bar"
          type="text"
        />
        <span>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-magnifying-glass "
            data-icon="magnifying-glass"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
              fill="currentColor"
            />
          </svg>
        </span>
        <button
          class="add-button"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-user-plus "
            data-icon="user-plus"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 640 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M96 128a128 128 0 1 1 256 0A128 128 0 1 1 96 128zM0 482.3C0 383.8 79.8 304 178.3 304l91.4 0C368.2 304 448 383.8 448 482.3c0 16.4-13.3 29.7-29.7 29.7L29.7 512C13.3 512 0 498.7 0 482.3zM504 312l0-64-64 0c-13.3 0-24-10.7-24-24s10.7-24 24-24l64 0 0-64c0-13.3 10.7-24 24-24s24 10.7 24 24l0 64 64 0c13.3 0 24 10.7 24 24s-10.7 24-24 24l-64 0 0 64c0 13.3-10.7 24-24 24s-24-10.7-24-24z"
              fill="currentColor"
            />
          </svg>
        </button>
      </div>
      <div
        class="phonebooks"
      />
    </section>
  </div>
</body>
`;
